controller/customerController.js

```diff
@@ -512,34 +589,51 @@ const updateCustomer = async (req, res) => {
       message: "Customer updated successfully!",
     });
   } catch (err) {
-    res.status(500).send({
-      message: err.message,
-    });
+    console.error("Error updating customer:", err);
+    res.status(500).send({ message: err.message });
   }
 };
 
 const deleteCustomer = (req, res) => {
-  Customer.deleteOne({ _id: req.params.id }, (err) => {
-    if (err) {
-      res.status(500).send({
-        message: err.message,
-      });
-    } else {
-      res.status(200).send({
-        message: "User Deleted Successfully!",
-      });
-    }
-  });
+  const { id } = req.params;
+  if (!mongoose.Types.ObjectId.isValid(id)) {
+    return res.status(400).send({ message: "Invalid customer ID" });
+  }
+
+  Customer.deleteOne(
+    { _id: new mongoose.Types.ObjectId(id) },
+    (err, result) => {
+      if (err) {
+        return res.status(500).send({ message: err.message });
+      }
+
+      if (result.deletedCount === 0) {
+        return res.status(404).send({ message: "Customer not found" });
+      }
+
+      res.status(200).send({ message: "User Deleted Successfully!" });
+    },
+  );
 };
 
 const getCustomerByEmail = async (req, res) => {
   try {
-    const customer = await Customer.findOne({ email: req.params.email });
+    const { email } = req.params;
+
+    if (
+      typeof email !== "string" ||
+      !/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)
```

Comment:
you are repeating this pattern it will be possible marked as duplicated, you should refactor that.